src/plugin.js

```diff
@@ -301,435 +264,65 @@ const contribAdsPlugin = function(options) {
+     *
+     * @param {Object} player The videojs player object
+     */
+    removeNativePoster() {
+      const tech = player.$('.vjs-tech');
+
+      if (tech) {
+        tech.removeAttribute('poster');
       }
     },
-    'content-playback': {
-      enter() {
-        // make sure that any cancelPlayTimeout is cleared
-        if (player.ads.cancelPlayTimeout) {
-          window.clearTimeout(player.ads.cancelPlayTimeout);
-          player.ads.cancelPlayTimeout = null;
-        }
 
-        // This was removed because now that "playing" is fixed to only play after
-        // preroll, any integration should just use the "playing" event. However,
-        // we found out some 3rd party code relied on this event, so we've temporarily
-        // added it back in to give people more time to update their code.
-        player.trigger({
-          type: 'contentplayback',
-          triggerevent: player.ads.triggerevent
-        });
-
-        // Play the content if cancelContentPlay happened and we haven't played yet.
-        // This happens if there was no preroll or if it errored, timed out, etc.
-        // Otherwise snapshot restore would play.
-        if (player.ads._cancelledPlay) {
-          if (player.paused()) {
-            player.play();
-          }
-        }
-      },
-      events: {
-        // In the case of a timeout, adsready might come in late.
-        // This assumes the behavior that if an ad times out, it could still
-        // interrupt the content and start playing. An integration could
-        // still decide to behave otherwise.
-        adsready() {
-          player.trigger('readyforpreroll');
-        },
-        adstart() {
-          this.state = 'ad-playback';
-          // This is a special case in which preroll is specifically set
-          if (player.ads.adType !== 'preroll') {
-            player.ads.adType = 'midroll';
-          }
-        },
-        contentupdate() {
-          if (player.paused()) {
-            this.state = 'content-set';
-          } else {
-            this.state = 'ads-ready?';
-          }
-        },
-        contentended() {
-
-          // If _contentHasEnded is true it means we already checked for postrolls and
-          // played postrolls if needed, so now we're ready to send an ended event
-          if (this._contentHasEnded) {
-            // Causes ended event to trigger in content-resuming.enter.
-            // From there, the ended event event is not redispatched.
-            // Then we end up back in content-playback state.
-            this.state = 'content-resuming';
-            return;
-          }
-
-          this._contentEnding = false;
-          this._contentHasEnded = true;
-          this.state = 'postroll?';
+    debug(...args) {
+      if (this.settings.debug) {
+        if (args.length === 1 && typeof args[0] === 'string') {
+          videojs.log('ADS: ' + args[0]);
+        } else {
+          videojs.log('ADS:', ...args);
```

Comment:
also is it actually necessary to split this up? won't this work for one argument?

Reply:
In safari it shows each argument on a different line, and this makes simple "log a string" cases look a lot better in safari.

Reply:
might make sense to use `String.concat` instead then 🤔

Reply:
They might not all be strings, though.

Reply:
`Array.join` perhaps?

Reply:
That would still do string concatenation and bypass the browser's special console output for non-string types.

Reply:
isn't that the same for `'ADS: ' + args[0]` ?

Reply:
It does a type check on line 304 to cover that.